### PR TITLE
[21.05] Restore Metadata size limit

### PR DESCRIPTION
**Pull request by mvdbeek**
Broke in https://github.com/galaxyproject/galaxy/pull/11902

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by bernt-matthias — APPROVED** [review]
Looks good to me. 

Was only wondering if the metadata is then set to `None`... (wrt our previous discussion `default` or `no_value` might be an option .. )

File: test/unit/data/test_metdata_limit.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@pytest.fixture(scope="module")
+def datatypes_registry():
+    r = registry.Registry()
+    r.load_datatypes()
+    set_datatypes_registry(r)
+
+
+@pytest.fixture
+def sa_session(datatypes_registry):
+    custom_types.MAX_METADATA_VALUE_SIZE = METADATA_LIMIT
```
Comment (nsoranzo) [inline test/unit/data/test_metdata_limit.py]: This can be moved outside of the function to line 12.

Comment (mvdbeek) [reply test/unit/data/test_metdata_limit.py]: Then it runs on import, I don't think that's an improvement ?

File: test/unit/data/test_metdata_limit.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import pytest
```
Comment (nsoranzo) [inline test/unit/data/test_metdata_limit.py]: The filename has a typo? `metdata` -> `metadata`

File: test/unit/data/test_metdata_limit.py (hunk 1 above)
Comment (nsoranzo) [reply test/unit/data/test_metdata_limit.py]: I don't think these are supposed to be imported, but we can leave it as it is just in case.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**